File: frontend/src/api/hermes-memory.ts
import { request } from './index'

export interface HermesProfileData {
  memory: string[]
  user: string[]
}

export interface HermesMemoryResponse {
  global: HermesProfileData
  profiles: Record<string, HermesProfileData>
}

export function fetchHermesMemory(): Promise<HermesMemoryResponse> {
  return request<HermesMemoryResponse>('/hermes-memory')
}

File: backend/app/models/hermes_memory.py
"""Pydantic models for hermes memory data."""

from pydantic import BaseModel, Field


class HermesProfileData(BaseModel):
    """Memory data for a single profile."""
    memory: list[str] = Field(default_factory=list)
    user: list[str] = Field(default_factory=list)


class HermesMemoryResponse(BaseModel):
    """Response for GET /api/hermes-memory."""
    global_data: HermesProfileData = Field(alias="global", default_factory=HermesProfileData)
    profiles: dict[str, HermesProfileData] = Field(default_factory=dict)

    class Config:
        populate_by_name = True
